cypher_character_frontend/src/main.rs

```diff
@@ -0,0 +1,151 @@
+use std::{borrow::Borrow, sync::Arc};
+
+use anyhow::Context;
+use askama::Template;
+use axum::{
+    extract::State, http::{header, StatusCode}, response::{Html, IntoResponse, Response}, routing::{get, post, put}, Form, Router
+};
+use cypher_character_model::Character;
+use cypher_character_model::Sentence;
+use serde::Deserialize;
+use tokio::sync::Mutex;
+use tracing::info;
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
+
+#[derive(Debug)]
+struct AppState {
+    character: Mutex<Character>,
+}
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    tracing_subscriber::registry()
+        .with(
+            tracing_subscriber::EnvFilter::try_from_default_env()
+                .unwrap_or_else(|_| "cypher_character_frontend=debug".into()),
+        )
+        .with(tracing_subscriber::fmt::layer())
+        .init();
+
+    
+
+    let app_state = Arc::new(AppState {
+        character: Mutex::new(Character {
+            name: "Tacos".to_string(),
+            pronouns: "yum/my".to_string(),
+            sentence: Sentence {
+                descriptor: "Delicious".to_string(),
+                character_type: "Avocado".to_string(),
+                focus: "Satiates the Hungry".to_string(),
+                flavor: Some("Spicy".to_string()),
+            },
+        }),
+    });
+
+    info!("initializing router...");
+    let api_router = Router::new()
+        .route("/v1/character", put(update_character));
+
+    let router = Router::new()
+        .nest("/api", api_router)
+        .route("/", get(character_sheet))
+        .route("/character", get(character_sheet))
+        .route("/character/edit", get(character_edit))
+        .route("/character/view", get(character_view))
+        .with_state(app_state);
+
+    let listener = tokio::net::TcpListener::bind("0.0.0.0:8000")
+        .await
+        .context("error creating listener")?;
+
+    info!("router initialized, now listening on port 8000");
+    axum::serve(listener, router)
+        .await
+        .context("error while starting server")?;
+
+    Ok(())
+}
+
+
+
+#[derive(Template)]
+#[template(path = "character/sheet.html")]
+struct CharacterSheetTemplate {
+    character: Character,
+}
+
+async fn character_sheet(State(state): State<Arc<AppState>>) -> CharacterSheetTemplate {
+
+    CharacterSheetTemplate {
+        character: state.character.lock().await.clone(),
```

Comment:
Potential future refactor: [`RwLock`](https://docs.rs/tokio/latest/tokio/sync/struct.RwLock.html) instead of a `Mutex`, since that is useful for the "many reads, few rights" case